parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetReader.java

```diff
@@ -104,7 +104,7 @@ public ParquetReader(Configuration conf, Path file, ReadSupport<T> readSupport,
     this(conf, file, readSupport, FilterCompat.get(unboundRecordFilter));
   }
 
-  private ParquetReader(Configuration conf,
+  public ParquetReader(Configuration conf,
```

Comment:
don't you want the other method on line 118?

Reply:
Yes, you are right. I need another method. Finally, I can get rid of the hack :) 

```
private val constructor = {
    val ctor = classTag.runtimeClass.getDeclaredConstructor(
      classOf[java.util.List[InputFile]],
      classOf[ParquetReadOptions],
      classOf[ReadSupport[Row]]
    )

    ctor.setAccessible(true)
    ctor
  }
  ```